### PR TITLE
add PCI(e) auto-IRQ detection and enable for NS16550 and Apollo Lake SoC

### PR DESCRIPTION
**Pull request by ghost**
This is a re-submission per Andy Ross's request that it be broken into discrete commits.

These patches add support for runtime determination of wired IRQs assigned to PCI devices, and implements that support in the NS16550 UART driver.  This functionality will be required more and more on the "big iron" Zephyr targets.  This is merely the groundwork.

This is also part of the larger job of refactoring the x86 board tree.

### CODE REVIEW

**Comment by gnuless** [comment]
I've submitted PR zephyrproject-rtos/ci-tools#58 to deal with the Kconfig issue.

**Comment by gnuless** [comment]
@andyross I've also corrected the template per your previous comments.

File: drivers/serial/CMakeLists.txt [hunk 1]
```diff
@@ -31,6 +30,19 @@ zephyr_library_sources_if_kconfig(uart_rv32m1_lpuart.c)
 
 zephyr_library_sources_ifdef(CONFIG_USERSPACE   uart_handlers.c)
 
+if(CONFIG_UART_NS16550)
+  zephyr_library_sources(uart_ns16550.c)
+  foreach(NUM RANGE 0 3)
+    if (CONFIG_UART_NS16550_PORT_${NUM})
+      configure_file(
+        uart_ns16550_port_x.h
```
Comment (andyross) [inline drivers/serial/CMakeLists.txt]: I didn't know this was a cmake thing.  Clever.

**Review by andyross — APPROVED** [review]

File: drivers/serial/CMakeLists.txt (hunk 1 above)
Comment (nashif) [reply drivers/serial/CMakeLists.txt]: indeed

**Review by nashif — APPROVED** [review]

**Pull request merged**